modules/ui/fields/input.js

```diff
@@ -153,7 +156,7 @@ export function uiFieldText(field, context) {
                         }
                         // make sure no extra decimals are introduced
                         const numDecimals = v.includes('.') ? v.split('.')[1].length : 0;
```

Comment:
This code added in f573c374c29901d6a481b9efbd3c6daa2f312aab is causing the funny behavior with the stepper buttons. `formatToParts` can identify the decimal part of the number more reliably; just need to extract that part from `floatParser`.